### PR TITLE
add missing typespecs

### PR DESCRIPTION
**Pull request by sasa1977**
These types are referenced in other typespecs in this project, but are not defined. 

Hope I've got the specs right :-)

### CODE REVIEW

File: lib/bson/types.ex [hunk 1]
```diff
@@ -181,6 +187,7 @@ defmodule BSON.JavaScript do
   Represents BSON JavaScript (with and without scope) types
   """
 
+  @type t :: %__MODULE__{code: binary, scope: binary}
```
Comment (michalmuskala) [inline lib/bson/types.ex]: I believe scope here is a map of `BSON.t` values

Comment (sasa1977) [reply lib/bson/types.ex]: Ah I see it now, good catch!

File: lib/mongo/cursor.ex [hunk 2]
```diff
@@ -8,6 +8,13 @@ import Record, only: [defrecordp: 2]
 defmodule Mongo.Cursor do
   @moduledoc false
 
+  @type t :: %__MODULE__{
+    conn: Mongo.conn,
+    coll: Mongo.collection,
+    query: BSON.document,
+    select: BSON.document | nil,
+    opts: Keyword.t
+  }
```
Comment (ericmj) [inline lib/mongo/cursor.ex:17]: The cursor modules are moduledoc private so I am thinking we should create a type `@opaque cursor :: Mongo.Cursor.t | Mongo.AggregationCursor.t | Mongo.SinglyCursor.t` on `Mongo` when referencing these types in public API.

Comment (sasa1977) [reply lib/mongo/cursor.ex:17]: Perhaps I'm missing something, but that's already done [here](https://github.com/ericmj/mongodb/blob/master/lib/mongo.ex#L52)?

**Comment by sasa1977** [comment]
Any idea why travis tests fail?

File: lib/mongo/cursor.ex (hunk 2 above)
Comment (ericmj) [reply lib/mongo/cursor.ex:17]: Oh. I guess it is.

**Comment by ericmj** [comment]
> Any idea why travis tests fail?

Something broke in the npm package we use to version mongodb.

File: lib/mongo/cursor.ex (hunk 2 above)
Comment (sasa1977) [reply lib/mongo/cursor.ex:17]: Yeah, that's how I descended down this rabbit hole. In our project, we're using dialyzer, and I tried to introduce the `--Wunknown` option, which warns on unknown types/functions. Given that the line I pointed to references non-existing types, I get some warnings.

**Pull request merged**